### PR TITLE
feature(release): add release merge wizard

### PR DESCRIPTION
**Pull request by lepokle**
This commit adds a merge wizard for releases. It is build upon the merge framework therefore
it works the same as the component and the vendor merge wizard: click merge on the target, select
a source and takeover the needed fields.

Merging releases is only possible inside a single component. In addition both releases need to have
at least on pair of equal source attachemnts nor no source attachments at all.

The fossology status is not merges since it will be superseeded by external tool requests in the
near future.

Signed-off-by: Leo von Klenze <leo.vonklenze@scansation.de>

### CODE REVIEW

**Comment by smrutis1** [comment]
Following observations found while testing the PR.
1) If a source release contains software platforms(viz: Java Platform, Micro Edition) in a comma separated value, while merging it's treated as two separate value.
2) If a source release contains linked releases with same name but different versions. while merging the version number is not showing, which is little confusing.

File: libraries/lib-datahandler/src/main/thrift/vulnerabilities.thrift [hunk 1]
```diff
@@ -246,4 +246,14 @@ service VulnerabilityService {
...
       **/
     VulnerabilityWithReleaseRelations getVulnerabilityWithReleaseRelationsByExternalId(1: string externalId, 2: User user);
+
+    /**
+     * returns a list of relations where the given release id is involved.
+     */
+    list<ReleaseVulnerabilityRelation> getReleaseVulnerabilityRelationsByReleaseId(1: string releaseId, 2: User user);
+
+    /**
+     * returns a lost of a allratings involving the given release id
```
[the remark this answers is not in the document]
Comment (lepokle) [reply libraries/lib-datahandler/src/main/thrift/vulnerabilities.thrift]: Done

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java [hunk 2]
```diff
@@ -880,6 +886,304 @@ public void recomputeReleaseDependentFields(Component component, String skipThis
...
+            ProjectReleaseRelationship relationship = project.getReleaseIdToUsage().remove(mergeSourceId);
+            // if the target release is also linked, keep this one, do not overwrite
+            if(!project.getReleaseIdToUsage().containsKey(mergeTargetId)) {
+                project.putToReleaseIdToUsage(mergeTargetId, relationship);
+            }
+            projectClient.updateProject(project, sessionUser);
+        }
+    }
+    
+    private void updateReleaseReferencesInAttachmentUsages(String mergeTargetId, String mergeSourceId, User sessionUser) throws TException {
```
[the remark this answers is not in the document]
Comment (lepokle) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java]: Done

**Comment by lepokle** [comment]
Hi @smrutis1 
1) This is not a problem of the merge wizard but of the handling of the field itself. Comma separated strings are immediately stored as separate values, even before the merge.
2) fixed

**Comment by smrutis1** [comment]
Tested again. Now the version numbers are showing for the linked releases.

**Pull request merged**